src/esp/gfx/CubeMap.h

```diff
@@ -0,0 +1,207 @@
+   * @param type, texture type
+   * @param imageFilePrefix, the filename prefix
+   * The 6 image files then would be:
+   * {imageFilePrefix}.{texType}.+X.png
+   * {imageFilePrefix}.{texType}.-X.png
+   * {imageFilePrefix}.{texType}.+Y.png
+   * {imageFilePrefix}.{texType}.-Y.png
+   * {imageFilePrefix}.{texType}.+Z.png
+   * {imageFilePrefix}.{texType}.-Z.png
+   * @return true, if success, otherwise false
+   *
+   * NOTE:
+   * In this version, it will lose precision when depth texture is saved to the
+   * disk. This is because the pixel format for depth texture is R32F with 1 bit
+   * of sign, 23 bits of mantissa and 8 bits of exponent while the radiance HDR
+   * is 8 bits of mantissa and 8 bits of exponent. So when it is saved, only 8
+   * bits is saved to the R channel (GB channels just repeat the R channel
+   * twice).
+   */
+  // TODO: save HDR textures in EXR format
+  bool saveTexture(TextureType type, const std::string& imageFilePrefix);
+#endif
+
+  /**
+   * ```
+   *           +----+
+   *           | -Y |
+   * +----+----+----+----+
+   * | -Z | -X | +Z | +X |
+   * +----+----+----+----+
+   *           | +Y |
+   *           +----+
+   * ```
+   * NOTE: +Y is top
+   * @brief load cubemap texture from external images
+   * @param type can be "rgba", "depth", or "objectId" (TODO)
+   * @param imageFilePrefix, the prefix of the image filename
+   * @param imageFileExtension, the image filename extension (such as "png",
+   * "jpg")
+   * @return true if succeeded, otherwise false
+   * The 6 image files then would be:
+   * {imageFilePrefix}.{texType}.+X.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.-X.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.+Y.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.-Y.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.+Z.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.-Z.{imageFileExtension}
+   */
+  void loadTexture(TextureType type,
+                   const std::string& imageFilePrefix,
+                   const std::string& imageFileExtension);
+
+  /**
+   * @brief Render to cubemap texture using the camera
+   * @param camera, a cubemap camera
+   * NOTE: It will NOT automatically generate the mipmap for the user
+   */
+  void renderToTexture(CubeMapCamera& camera,
+                       scene::SceneGraph& sceneGraph,
+                       RenderCamera::Flags flags);
+
+ private:
+  Flags flags_;
+  int imageSize_ = 0;
+  std::map<TextureType, std::unique_ptr<Magnum::GL::CubeMapTexture>> textures_;
+
+  /**
+   * @brief Recreate textures
+   */
+  void recreateTexture();
+
+  // framebuffers (one for every cube side)
+  Corrade::Containers::StaticArray<6, Magnum::GL::Framebuffer> frameBuffer_{
+      Corrade::Containers::DirectInit, Magnum::NoCreate};
+
+  // in case there is no need to output depth texture, we need a depth buffer
+  Magnum::GL::Renderbuffer optionalDepthBuffer_[6];
```

Comment:
`NoCreate` those as well? Otherwise it unnecessarily creates 6 GL renderbuffers even if not needed.